### DISCUSSION TITLE
Create new state property for vacuum which is using an enum

### DISCUSSION THREAD

**Discussion by gjohansson-ST** [home-assistant/architecture#1141] [Entity Models] [answered]
## Background

This is more or less a copy of #1140 but targets `vacuum`

Some entity models (like `vacuum`) allows integration to directly overwrite `state` property to set it's state using constants and are not modeled like the newer ones (as example `lawn_mower) which has a separate property for handling state using an enum.

## Proposal

Similar to the `lawn_mower` integration implement a separate state property `activity` which is using an enum `VacuumActivity` for controlling it's state.

```python
@final
@cached_property
def state(self) -> str | None:
    """Return the activity of the vacuum cleaner."""
    return activity

@cached_property
def activity(self) -> VacuumActivity | None:
    """Return the current vacuum entity state.

    Integrations should overwrite this or use the '_attr_activity'
    attribute to set the vacuum state using the 'VacuumActivity' enum.
    """
    return self._attr_activity
```

After the deprecation period integrations should only implement `activity` returning `VacuumActivity`.

Draft conceptual PR:https://github.com/home-assistant/core/pull/126353

### Alternatives to consider

Initial proposal was to add `vacuum_state`, has been updated to be modeled similar to `lawn_mover` using `activity` instead.
- Perhaps (similar to `lawn_mower`) the state property for a vacuum should be `activity` and the enum would be called `VacuumActivity`

## Update after initial proposal

- Return the enum member instead of stringifying it's state

**Comment by MartinHjelmare** [comment]
It would be nice to make it similar to lawnmower although as for lawnmower some of the enum values don't sound as activities, eg error. Maybe not a big deal, and it's more important to leverage consistency?

**Reply by gjohansson-ST** [reply]
Same for vacuum and the state/activity is a mix of activity and state.
But I think given the two are almost the same, that we should model them similar.
There is less point in consistency between this and, as example, `alarm_control_panel` which is the other open proposal.
I will update the proposal.

**Comment by MartinHjelmare** [comment, marked as answer]
We've discussed this proposal in the Core team and decided to approve it.

**Reply by gjohansson-ST** [reply]
Closing as it has been implemented per core PR mentioned in description
